platform-tools-sdk/cargo-build-sbf/src/post_processing.rs

```diff
@@ -1,63 +1,176 @@
 use {
-    crate::{spawn, toolchain::rust_target_triple, Config},
+    crate::{
+        spawn,
+        toolchain::rust_target_triple,
+        utils::{copy_file, create_directory, generate_keypair},
+        Config,
+    },
     log::{debug, error, info, warn},
     regex::Regex,
-    solana_keypair::{write_keypair_file, Keypair},
     std::{
         collections::{HashMap, HashSet},
         fs::{self, File},
         io::{BufRead, BufReader, BufWriter, Write},
-        path::Path,
+        path::{Path, PathBuf},
         process::exit,
         str::FromStr,
     },
 };
 
+fn file_older_or_missing(prerequisite_file: &Path, target_file: &Path) -> bool {
+    let prerequisite_metadata = fs::metadata(prerequisite_file).unwrap_or_else(|err| {
+        error!(
+            "Unable to get file metadata for {}: {}",
+            prerequisite_file.display(),
+            err
+        );
+        exit(1);
+    });
+
+    if let Ok(target_metadata) = fs::metadata(target_file) {
+        use std::time::UNIX_EPOCH;
+        prerequisite_metadata.modified().unwrap_or(UNIX_EPOCH)
+            > target_metadata.modified().unwrap_or(UNIX_EPOCH)
+    } else {
+        true
+    }
+}
+
+fn create_folders(deploy_folder: &PathBuf, debug_folder: &PathBuf) {
+    if !deploy_folder.exists() {
+        create_directory(deploy_folder);
+    }
+
+    if !debug_folder.exists() {
+        create_directory(debug_folder);
+    }
```

Comment:
What do you think about only creating this folder if `config.debug` is set? That way we don't needlessly create directories and potentially confuse people